File: arrays/readability.cpp
//1. get text
#include <iostream>

int main(void)
{
  
  std::string text = "";
  std::cout << "text: ";
  std::getline(std::cin, text);
  int sentences = 0, letters = 0, words = 0;
  bool in_word = false;
  for(char t : text)
  {
    char i = tolower(t);
    if(in_word && (i == ' ' || i == '\n' || i == '\t'))
    {
      words++;
      in_word = false;
    }
    else if(i == '.' || i == '?' || i == '!')
    {
        sentences++;
    }
    else if(i != ' ' && i >= 'a' && i <= 'z')
    {
      letters++;
      in_word = true;
    }
  }
  if(in_word) words++;
  if(words == 0)
  {
    std::cout << "no text entered!\n";
    return 0;
  }
  std::cout << letters << "\n" << sentences << "\n";
  double L = 100 * letters / words;
  double S = 100 * sentences / words;
  double index = 0.0588 * L - 0.296 * S - 15.8;
  std::string grade = (index < 1) ? "<1" : ((index > 16) ? "16+" : std::to_string(int(round(index))));
  std::cout << index << "\ngrade: " << grade << "\n";
}

File: arrays/readability/readability.py
text = input("text: ")
letters, words, sentences = [0, 0, 0]
in_word = False
for txt in text.lower():
  t = ord(txt)
  if(in_word and (txt == ' ' or txt == '\n' or txt == '\t')):
    words += 1
    in_word = False
  if(txt != ' ' and t >= ord('a') and t <= ord('z')):
    letters += 1
    in_word = True
  if(txt == '.' or txt == '!' or txt == '?'):
    sentences += 1

words += 1 if in_word else 0
if(words == 0):
  print('no text entered!')

else:
  L = 100 * letters / words
  S = 100 * sentences / words

  index = 0.0588 * L - 0.296 * S - 15.8;
  grade = '16+' if (index > 16) else ( '<1' if (index < 1) else str(int(round(index))))
  print(f'grade : {grade}')
